### PR TITLE
ELIG-3368-PARENT-FRONT-END-Implement-design-changes-to-show-correlation-ID-and-error-code-to-user-for-single-check-technical-error-outcomes

### PR DESCRIPTION
**Pull request by J-C-Barton**
- Add Error Code and Correlation ID to the Technical_Error outcome for single checks.

https://dfedigital.atlassian.net/browse/ELIG-3368

### CODE REVIEW

File: CheckYourEligibility.FrontEnd.Tests/Controllers/CheckControllerTests.cs [hunk 1]
```diff
@@ -703,6 +704,26 @@ public async Task Given_Loader_When_StatusProvided_Should_ReturnCorrectView(stri
         _getCheckStatusUseCaseMock.Verify(x => x.Execute(responseJson, _sessionMock.Object), Times.Once);
     }
 
+    public async Task Given_Loader_When_Status_TechnicalError_Should_ReturnErrorCode()
```
Comment (YiannosGeorgantas) [inline CheckYourEligibility.FrontEnd.Tests/Controllers/CheckControllerTests.cs:708]: The new Given_Loader_When_Status_TechnicalError_Should_ReturnErrorCode method is missing the NUnit [Test] attribute, so it will not be discovered or executed. Please add [Test] here. The same issue appears in all three ELIG-3368 PRs.

File: CheckYourEligibility.FrontEnd.Tests/Controllers/CheckControllerTests.cs [hunk 2]
```diff
@@ -703,6 +704,26 @@ public async Task Given_Loader_When_StatusProvided_Should_ReturnCorrectView(stri
...
+
+        _getCheckStatusUseCaseMock
+            .Setup(x => x.Execute(responseJson, _sessionMock.Object))
+            .ReturnsAsync(response.Data);
+        // Act
+        var result = await _sut.Loader();
+
+        // Assert
+        var viewResult = result as ViewResult;
+        viewResult.ViewData["ErrorCode"].Should().Be(response.Data.ErrorCode);
```
Comment (YiannosGeorgantas) [inline CheckYourEligibility.FrontEnd.Tests/Controllers/CheckControllerTests.cs:732]: Could this test cover the complete technical-error behaviour rather than only ErrorCode? Please also assert that the technical-error view is returned and that the correlation ID is passed through correctly. We should also have coverage for the requirement that the correlation ID is omitted when unavailable, particularly while the Cypress test remains excluded from CI.

File: CheckYourEligibility.FrontEnd/Views/Check/Outcome/Technical_Error.cshtml [hunk 3]
```diff
@@ -12,17 +12,29 @@
...
+                    <p>
+                        <a href="~/Check/Enter_Details">Try again</a>, ensuring you’ve entered your details accurately.
                     </p>
                 </div>
             </div>
 
-            <h1 class="govuk-heading-l">If the check keeps failing</h1>
+            <h2>How to report this error</h2>
+            <p>Copy the following details and send them to our team using the <a target="_blank" href="https://customerhelpportal.education.gov.uk/">Customer Help Portal</a>.</p>
+            @if (correlationID != null)
```
Comment (YiannosGeorgantas) [inline CheckYourEligibility.FrontEnd/Views/Check/Outcome/Technical_Error.cshtml]: The requirement says the correlation ID should be omitted when no GUID is available. Could this use !string.IsNullOrWhiteSpace(correlationID) rather than checking only for null? Otherwise an empty value would still display a blank Correlation ID: row.

File: CheckYourEligibility.FrontEnd.Tests/Controllers/CheckControllerTests.cs (hunk 1 above)
Comment (J-C-Barton) [reply CheckYourEligibility.FrontEnd.Tests/Controllers/CheckControllerTests.cs:708]: Done.

File: CheckYourEligibility.FrontEnd.Tests/Controllers/CheckControllerTests.cs (hunk 2 above)
Comment (J-C-Barton) [reply CheckYourEligibility.FrontEnd.Tests/Controllers/CheckControllerTests.cs:732]: I have updated the test. If the correlationID is omitted the view handles that rather than the controller so I don't think a unit test is worth implementing for that scenario. It's only putting a value (or not) in ViewData. We can discuss further or raise in Dev Huddle if you think it's important though?

File: CheckYourEligibility.FrontEnd/Views/Check/Outcome/Technical_Error.cshtml (hunk 3 above)
Comment (J-C-Barton) [reply CheckYourEligibility.FrontEnd/Views/Check/Outcome/Technical_Error.cshtml]: Done.

File: CheckYourEligibility.FrontEnd/Usecases/GetCheckStatusUseCase.cs [hunk 4]
```diff
@@ -40,9 +40,14 @@ public async Task<string> Execute(string responseJson, ISession session)
             throw new Exception("Null response received from GetStatus.");
         }
 
+        if (response.Links.Get_EligibilityCheck != null)
```
Comment (YiannosGeorgantas) [inline CheckYourEligibility.FrontEnd/Usecases/GetCheckStatusUseCase.cs:43]: Having now traced this through `CheckGateway.GetStatus`, the gateway also dereferences `responseBody.Links.Get_EligibilityCheck`, so my original suggested guard would have been too late.

Could the missing-link case instead be handled in `GetCheckStatusUseCase` before `_checkGateway.GetStatus(response)` is called? When the initial response contains the technical-error status and error code but no eligibility-check link, the use case should return `response.Data` directly, leaving `CorrelationID` unset. This preserves the error code for the technical-error view without attempting a status request that cannot be made.

For example, the flow could be equivalent to:

```csharp
var eligibilityCheckLink = response?.Links?.Get_EligibilityCheck;

if (string.IsNullOrEmpty(eligibilityCheckLink))
{
    session.SetString("CheckResult", response.Data.Status);
    return response.Data;
}

var check = await _checkGateway.GetStatus(response);

// Existing null-response handling...

check.Data.CorrelationID =
    eligibilityCheckLink.Replace("/check/", "");

return check.Data;
```

Could we also add a unit test using an error response with `Links = null`, asserting that the gateway is not called, the error code is returned, and the correlation ID remains unset?

**Review by YiannosGeorgantas — CHANGES_REQUESTED** [review]
Thanks for addressing the previous comments. The updated controller test now covers the technical-error view, error code and correlation ID correctly. I’ve left one further comment regarding the case where the response contains no `Links` value, as this can currently prevent the available error code from reaching the technical-error view. Once that is handled and covered, I’m happy to approve.

File: CheckYourEligibility.FrontEnd/Usecases/GetCheckStatusUseCase.cs (hunk 4 above)
Comment (YiannosGeorgantas) [reply CheckYourEligibility.FrontEnd/Usecases/GetCheckStatusUseCase.cs]: Following our discussion, I’m withdrawing this additional request. The existing technical-error page is the agreed scope; submission-failure routing is outside this change.

**Review by YiannosGeorgantas — APPROVED** [review]
